hooks/fetch/use-API-route-data.ts

```diff
@@ -0,0 +1,55 @@
+import { useEffect, useState } from 'react';
+import { APILoadingFactory, IAPILoading } from '#models/api-loading';
+import { httpGet } from '#utils/network';
+import {
+  FailToFetchError,
+  RequestAbortedDuringUnloadException,
+} from '#utils/network/frontend';
+import logErrorInSentry from '#utils/sentry';
+import { APIPath, RouteResponse } from 'app/api/data-fetching/routes';
+
+/**
+ * Hook to fetch data from internal API
+ * @param route : route to fetch (from {@link APIPath})
+ * @param slug : slug parameter for the route
+ * @returns {IAPILoading | RouteResponse<T>} - The API loading state or the fetched data
+ */
+export function useAPIRouteData<T extends APIPath>(
+  route: T,
+  slug: string
+): IAPILoading | RouteResponse<T> {
+  const [response, setResponse] = useState<IAPILoading | RouteResponse<T>>(
+    APILoadingFactory()
+  );
```

Comment:
Dont we introduce scope checking in front with a `IUnauthorised` ? Or is `DataSection` not ready for this refacto yet ?